src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -200,5 +216,27 @@ void ResourceAllocationManagerImpl::OnDriverDisallowed(
   list_of_rejected_resources.push_back(module_type);
 }
 
-ResourceAllocationManager::~ResourceAllocationManager() {}
+void ResourceAllocationManagerImpl::RefreshAllocatedResourcesState() {
+  LOG4CXX_AUTO_TRACE(logger_);
```

Comment:
@AKalinich-Luxoft 
Suggest to investigate approach when AM notifies plugin on application unregistration. There is a similar implementation for OnDeviceRemoved notification. Otherwise you'll have to not forget to do manually that cleanup in every other place/class.